tests/components/schlage/test_sensor.py

```diff
@@ -0,0 +1,30 @@
+"""Test schlage sensor."""
+
+from homeassistant.components.sensor import SensorDeviceClass
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+
+async def test_sensor_device_registry(
+    hass: HomeAssistant, mock_added_config_entry: ConfigEntry
+) -> None:
+    """Test sensor is added to device registry."""
+    device_registry = dr.async_get(hass)
+    device = device_registry.async_get_device(identifiers={("schlage", "test")})
+    assert device.model == "<model-name>"
+    assert device.sw_version == "1.0"
+    assert device.name == "Vault Door"
+    assert device.manufacturer == "Schlage"
+
+
+async def test_battery_sensor(
+    hass: HomeAssistant, mock_added_config_entry: ConfigEntry
+) -> None:
+    """Test the battery sensor."""
+    battery_sensor = hass.states.get("sensor.vault_door")
```

Comment:
I think this should come out as `sensor.vault_door_battery` if it's correctly done?

Reply:
Yep. If I had actually run the tests, I would have seen this. 🤦🏻‍♂️ ]

Fixed.